File: test/python/bench/fib_recursive.py
"""Mirror of test/jacl/bench/fib_recursive.jacl.

Naive recursive Fibonacci. fib(25) = 75025; ~242 k calls, max stack
depth 25, no allocation in the hot path.
"""


def fib(n: int) -> int:
    if n < 2:
        return n
    return fib(n - 1) + fib(n - 2)


def run() -> int:
    return fib(25)


if __name__ == "__main__":
    print(run())

File: test/clojure/bench/fib_recursive.clj
;; Mirror of test/jacl/bench/fib_recursive.jacl.
;;
;; Naive recursive Fibonacci. fib(25) = 75025; ~242 k calls, max stack
;; depth 25. Exercises function-call dispatch (Var deref + invoke) and
;; integer arithmetic.

(ns bench.fib-recursive)

(defn fib [n]
  (if (< n 2)
    n
    (+ (fib (- n 1)) (fib (- n 2)))))

(defn run [] (fib 25))
